AK/StringUtils.h

```diff
@@ -38,6 +39,21 @@ enum class TrimWhitespace {
     No,
 };
 
+enum class SplitBehavior : unsigned {
+    // Neither keep empty substrings nor keep the trailing separator.
+    // This is the default behavior if unspecified.
+    Nothing = 0,
+
+    // If two separators follow each other without any characters
+    // in between, keep a "" in the resulting vector. (or only the
+    // separator if KeepSeparator is used)
+    KeepEmpty = 1,
+
+    // Do not strip off the separator at the end of the string.
+    KeepSeparator = 2,
```

Comment:
I think this should be renamed, because it could be any of these three options:
- `"a,b,c" -> "a," ",b," ",c"
- `"a,b,c" -> "a," "b," "c"
- `"a,b,c" -> "a" ",b" ",c"

And how does it work without keepempty?

- "a,,,b" -> "a," "b"
- "a,,,b" -> "a," ",b"
- "a,,,b" -> "a," "," "," "b"
- "a,,,b" -> "a," "," ",b"

To me it feels like KeepSeparator is something the caller can do better and (except maybe for the callback) It is vague

Reply:
I agree that this is ambiguous, I renamed it to `KeepTrailingSeparator`. Without KeepEmpty, it does not keep the separator (otherwise, there would be no point in being able to specify KeepEmpty with KeepTrailingSeparator). I also added some tests.